src/Nav.tsx

```diff
@@ -28,6 +27,30 @@ import AuthorJpg from "./images/nav/author.jpg";
 import SteamJpg from "./images/nav/steam.jpg";
 
 import SourceIcon from "@mui/icons-material/Source";
+import { OverridableComponent } from "@mui/material/OverridableComponent";
+
+class RouteData {
+  public url: string;
+  public title: string;
+  public iconComponent:
+    | string
+    | OverridableComponent<SvgIconTypeMap<{}, "svg">>;
+  constructor(data: RouteData) {
+    this.url = data.url;
+    this.title = data.title;
+    this.iconComponent = data.iconComponent;
+  }
+}
+
+function isstring(arg: any): arg is string {
```

Comment:
This function is only used once is it really needed?

Reply:
typeof(v) === "aType" is also really common so people should just understand what is going on by seeing it.